src/contracts/pods/EigenPodManager.sol

```diff
@@ -134,68 +202,215 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
+                middlewareTimesIndex
+            ),
+            "EigenPodManager.exitUndelegationLimbo: shares in limbo are still slashable"
+        );
+
+        require(limboStartBlock + strategyManager.withdrawalDelayBlocks() <= block.number,
+            "EigenPodManager.exitUndelegationLimbo: withdrawalDelayBlocks period has not yet passed"
+        );
+
+        // delete the pod owner's undelegation limbo details
+        delete _podOwnerUndelegationLimboStatus[msg.sender];
+
+        // emit event
+        emit UndelegationLimboExited(msg.sender);
+
+        // either withdraw the funds entirely from EigenLayer
+        if (withdrawFundsFromEigenLayer) {
+            // store the pod owner's current share amount in memory, for gas efficiency
+            uint256 currentPodOwnerShares = podOwnerShares[msg.sender];
+            // remove the shares from the podOwner and reduce the `withdrawableRestakedExecutionLayerGwei` in the pod
+            podOwnerShares[msg.sender] = 0;
+            _decrementWithdrawableRestakedExecutionLayerGwei(msg.sender, currentPodOwnerShares);
+            // withdraw through the ETH from the EigenPod
+            _withdrawRestakedBeaconChainETH(msg.sender, msg.sender, currentPodOwnerShares);
+        // or else return the "shares" to the delegation system
+        } else {
+            delegationManager.increaseDelegatedShares(msg.sender, beaconChainETHStrategy, podOwnerShares[msg.sender]);
+        }
+    }
+    
+    // EXTERNAL FUNCTIONS PERMISSIONED TO SINGLE PARTIES
+    // TODO: write documentation for this function
+    function slashShares(
+        address slashedPodOwner,
+        address slashedFundsRecipient,
+        uint256 shareAmount
+    )
+        external
+        onlyOwner
+        onlyFrozen(slashedPodOwner)
+        nonReentrant
     {
-        ownerToPod[podOwner].incrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        require(
+            // either the pod owner themselves is frozen
+            slasher.isFrozen(slashedPodOwner) ||
+            // or they are in "undelegation limbo" and the operator who they *were* delegated to is frozen
+            (
+                isInUndelegationLimbo(slashedPodOwner) &&
+                slasher.isFrozen(_podOwnerUndelegationLimboStatus[slashedPodOwner].delegatedAddress)
+            ),
+            "EigenPodManager.slashShares: cannot slash the specified pod owner"
+        );
+        require(shareAmount > 0, "EigenPodManager.slashShares: shares must be greater than zero");
+        require(podOwnerShares[slashedPodOwner] >= shareAmount,
+            "EigenPodManager.slashShares: podOwnerShares[podOwner] must be greater than or equal to shares");
+
+        _removeShares(slashedPodOwner, shareAmount);
+        // TODO: @Sidu28 -- confirm that decrementing `withdrawableRestakedExecutionLayerGwei` is correct/intended here
+        _decrementWithdrawableRestakedExecutionLayerGwei(slashedPodOwner, shareAmount);
```

Comment:
there's a TODO here in the code for @Sidu28

Reply:
actually nevermind, I have a PR open to remove this function entirely, should probably just merge that first